### PR TITLE
Add note about EnableNETAnalyzers when NuGet packaged is installed

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #23010

### CODE REVIEW

**Review by mavasani — APPROVED** [review]

File: docs/core/project-sdk/msbuild-props.md [hunk 1]
```diff
@@ -404,6 +404,9 @@ The `CodeAnalysisTreatWarningsAsErrors` property lets you configure whether code
 </PropertyGroup>
 ```
 
+> [!NOTE]
+> This property applies specifically to the built-in analyzers in the .NET 5+ SDK. It should not be used when you install a NuGet code analysis package.
```
Comment (Youssef1313) [inline docs/core/project-sdk/msbuild-props.md:408]: When I have the NuGet package and also not specifying EnableNETAnalyzers, there is still a warning if I'm using an older package than what comes with the SDK:

>  warning : The .NET SDK has newer analyzers with version '6.0.0' than what version '5.0.3' of 'Microsoft.CodeAnalysis.NetAnalyzers' package provides. Update or remove this package reference.

Is there a way to pin to a specific version without getting warnings? (which usually should fail CI)

File: docs/fundamentals/code-analysis/overview.md [hunk 2]
```diff
@@ -91,7 +91,10 @@ By default, you'll get the latest code analysis rules and default rule severitie
 
   For more information, and to see a list of possible values, see [AnalysisLevel](../../core/project-sdk/msbuild-props.md#analysislevel).
 
-- Install the [Microsoft.CodeAnalysis.NetAnalyzers NuGet package](https://github.com/dotnet/roslyn-analyzers#microsoftcodeanalysisnetanalyzers) to decouple rule updates from .NET SDK updates. Installing the package turns off the built-in SDK analyzers and generates a build warning if the SDK contains a newer analyzer assembly version than that of the NuGet package.
+- Install the [Microsoft.CodeAnalysis.NetAnalyzers NuGet package](https://github.com/dotnet/roslyn-analyzers#microsoftcodeanalysisnetanalyzers) to decouple rule updates from .NET SDK updates. For projects that target .NET 5+, installing the package turns off the built-in SDK analyzers and generates a build warning if the SDK contains a newer analyzer assembly version than that of the NuGet package.
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/overview.md:94]: > For projects that target .NET 5+, installing the package turns off the built-in SDK analyzers and generates a build warning if the SDK contains a newer analyzer assembly version than that of the NuGet package.

This warning is produced regardless the TFM based on what I've tried. The current wording implies the warning is for net5.0+ TFM only.

File: docs/fundamentals/code-analysis/overview.md [hunk 3]
```diff
@@ -91,7 +91,10 @@ By default, you'll get the latest code analysis rules and default rule severitie
...
+
+  > [!NOTE]
+  > If you install the Microsoft.CodeAnalysis.NetAnalyzers NuGet package, you should not add the [EnableNETAnalyzers](../../core/project-sdk/msbuild-props.md#enablenetanalyzers) property to either your project file or a *Directory.Build.props* file. When the NuGet package is installed, the presence of this MSBuild property will generate a build warning.
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/overview.md:97]: > you should not add the [EnableNETAnalyzers property

Should this read "you should not set EnableNETAnalyzers property to true" instead?

**Review by Youssef1313 — APPROVED** [review]
Looks great with minor comments.

I have to say that I love how the code-analysis docs have evolved and how you manage to make it clear, informative, and to the point. This is really great :heart: :tada:

**Comment by gewarren** [comment]
> Looks great with minor comments.
> 
> I have to say that I love how the code-analysis docs have evolved and how you manage to make it clear, informative, and to the point. This is really great ❤️ 🎉

Thank you for the nice feedback :D

File: docs/core/project-sdk/msbuild-props.md (hunk 1 above)
Comment (gewarren) [reply docs/core/project-sdk/msbuild-props.md:408]: @mavasani Should we advertise the `<_SkipUpgradeNetAnalyzersNuGetWarning>` property? It seems to defeat the purpose of the design of generating a warning when the SDK contains newer analyzers. If people want to decouple analyzer updates from SDK updates, then should we really generate a warning, or should it just be a suggestion?

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

You can :shipit: when ready.

File: docs/core/project-sdk/msbuild-props.md (hunk 1 above)
Comment (gewarren) [reply docs/core/project-sdk/msbuild-props.md:408]: I created a new issue to track that.

**Pull request merged**